google/cloud/datastore/query.py

```diff
@@ -209,30 +329,62 @@ def filters(self):
         """
         return self._filters[:]
 
-    def add_filter(self, property_name, operator, value):
+    def add_filter(
+        self,
+        property_name=None,
+        operator=None,
+        value=None,
+        *,
+        composite_filter=None,
```

Comment:
Why are there separate parameters for whether this is a property filter or a composite filter?

Reply:
I was thinking of making it explicit that we can accept two kinds of filter: a `CompositeFilter` or a `PropertyFilter`.
After our offline chat about its, I think it would be better if we accept just one parameter `filter` which can any descendants of the `BaseFilter` class.

I will make this change so that we will accept only one optional parameter (`filter`) instead of `property_filter` and `composite_filter`. This way, we can also remove the check below (https://github.com/googleapis/python-datastore/pull/418/files#diff-6053552b52329f41530b743e3fcb37f9e7622c7e165f751b94d08174df79ac26R442) where it raises error if user tries to pass both paramaters at the same time.

Thanks.

Reply:
I have replaced the two parameters so there is only a `filter` parameter here now.